src/System.Net.Mail/src/System/Net/Mail/NTAuthenticationStub.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Net.Security;
+using System.Security.Authentication.ExtendedProtection;
+using System.Diagnostics;
+
+namespace System.Net
+{
+    internal class NTAuthentication
```

Comment:
> Just curious: Is this stub temporary due to layering issues that will be resolved before shipping, or is it going to ship this way?

This work is similar to AuthenticationManager stubbed out in [here](https://github.com/dotnet/corefx/blob/2117bae3a6c9736033e6d1659821ff3a1069920b/src/System.Net.Requests/src/System/Net/AuthenticationManager.cs). 

Ideally SMTP should use AuthenticationManager, and use the RegisteredModules for authentication, since that is stubbed out in .NET Core, and we need to provide authentication, using Reflection for now. This work will be related to how the AuthenticationManger is solved, either by merging libs, or exposing new APIs, and out of scope for this release.

cc @CIPop @weshaggard Can confirm.